### DISCUSSION TITLE
Checking that  token is expired

### DISCUSSION THREAD

**Discussion by fade2black** [jwt/ruby-jwt#433] [Ideas]
Hello,
Why not add a new method like `JWT.expired?(token)`? Otherwise we have to decode the token which raises exception. But with such a method we could check from anywhere whether or not a token is expired. I already read this discussion #124.

**Comment by excpt** [comment]
Thinking about Version 3.0. 

Changing the api to use methods like `.decode`, `.decode!`, `.encode` and `.encode!`.

That gives us the space to implement a method like `.expired?()` or `.valid?()` on the `.encode()` or `.decode()` results.

**Reply by fade2black** [reply]
@excpt Is work in progress, or can I give you a hand?

**Comment by anakinj** [comment]
The new JWT objects should give some flexibility on how to check the validity of the tokens.
